api/src/main/java/org/eclipse/microprofile/openapi/annotations/PathItemOperation.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.openapi.annotations;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.eclipse.microprofile.openapi.annotations.callbacks.Callback;
+import org.eclipse.microprofile.openapi.annotations.extensions.Extension;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirementsSet;
+import org.eclipse.microprofile.openapi.annotations.servers.Server;
+
+/**
+ * Describes an Operation
+ * <p>
+ * This annotation is only used for operations defined under a {@link PathItem}. For operations provided by the API
+ * itself, it's more common to use the {@link Operation} annotation applied to a Jakarta REST resource method instead.
+ *
+ * @since 4.0
+ */
+@Target({})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface PathItemOperation {
+
+    /**
+     * The HTTP method for this operation.
+     *
+     * @return the HTTP method of this operation
+     **/
+    String method();
+
+    /**
+     * The tag names which apply to this operation.
+     *
+     * @return the list of tag names
+     */
+    String[] tags() default {}; // Could be @Tag[]
+
```

Comment:
I remembered there was one bit I wasn't sure about and it was this.

At the moment, we allow `@Tag` to be put on a rest resource method and it will tag the method and create a tag entry at the top level. The user can put a description for the tag in there as well, but if they put a different description in different places then it's undefined which one ends up in the final document.

We could replicate that function here, or we could require the user to just use the tag name, and use the `@Tag` annotation separately if they want a description.

Reply:
Decided to use the `@Tag` annotation here. It's a bit more verbose but it's consistent with how tags are added to operations elsewhere.

I've done it as a separate commit so it can be easily removed if needed.

Reply:
+1 to how you've done it.